Boto vpc peering fix

blarghmatey opened the pull request:
### What does this PR do?

Fix a bug in the keyword arguments used for `is_peering_connection_pending` in the state function `accept_vpc_peering_connection`.
### What issues does this PR fix or reference?

None
### Previous Behavior

When using the `accept_vpc_peering_connection` state function it would result in the following stacktrace:

```
               An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1744, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1704, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/boto_vpc.py", line 1487, in accept_vpc_peering_connection
                  profile=profile
              TypeError: is_peering_connection_pending() got an unexpected keyword argument 'name'
```
### New Behavior

An exception is not raised.
### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

xiaoanyunfei committed:
fix salt-api opts

xiaoanyunfei committed:
Merge branch '2016.3' of github.com:saltstack/salt into 2016.3

xiaoanyunfei committed:
Merge branch '2016.3' of github.com:saltstack/salt into 2016.3

xiaoanyunfei committed:
add simplify code

xiaoanyunfei committed:
move back

UtahDave committed:
If windows pkg db hasn't been created yet, refresh the db instead of stacktracing (#36008)

* Don't raise an exception. Just return empty dict

If the repo cache file doesn't exist, lets' just return
an empty dict rather than raising an exception.

* If repo cache file doesn't exist, refresh

* add logging message

* remove unused imports

Commit:
Back-port #36070 to 2015.8 (#36169)

twangboy committed:
Fix test_launchctl test

twangboy committed:
Fix unquoted integers

Commit:
Don't run the same signal handler twice. Catch os.kill errors.

Else a Process Not Found error stops the killing loop and there are
still alive master subprocesses.

Commit:
Removed unused import.

twangboy committed:
Fix some lint

t2b committed:
Allow additional kwargs in states.dockerng.image_present (#36156)

Fixes #31513

Commit:
Merge branch '2015.8' into '2016.3'

Conflicts:
  - salt/cloud/clouds/opennebula.py

Commit:
Merge pull request #36188 from rallytime/merge-2016.3

[2016.3] Merge forward from 2015.8 to 2016.3

thatch45 committed:
if the backend stack traces when it should return an empty string (#36193)

xiaoanyunfei committed:
Merge branch '2016.3' of github.com:saltstack/salt into 2016.3

xiaoanyunfei committed:
fix log owner

xiaoanyunfei committed:
fix logqueue owner

xiaoanyunfei committed:
Merge branch 'logowner' of https://github.com/xiaoanyunfei/salt into logowner

xiaoanyunfei committed:
cancle pr last

Commit:
Merge pull request #36203 from xiaoanyunfei/logowner

fix owner of MultiprocessingLoggingQueue

Commit:
Merge pull request #36184 from DSRCorporation/bugs/35480_master_shutdown

Disable signal handling while handling signal

terminalmage committed:
Add documentation for fileserver_list_cache_time

terminalmage committed:
Add function to remove fileserver list caches to fileserver class

terminalmage committed:
Add runner function to remove file list caches

terminalmage committed:
Couple fixes to _gen_back

1. Wasn't properly handling the backend being passed in as a Python list
2. Didn't work properly in the test suite due to the backend being an
   ImmutableList instead of a list. This is because of how the opts are
   loaded in the test suite, all mutable types are immutable versions of
   themselves from salt.utils.immutabletypes.

terminalmage committed:
Add integration tests for fileserver.clear_file_list_cache

Commit:
Docs clarification for module sync and state.apply (#36217)

Closes #35340

Commit:
Merge branch '2016.3' into 'carbon'

Conflicts:
  - salt/modules/win_pkg.py
  - salt/states/dockerng.py
  - salt/utils/process.py

blarghmatey committed:
Fixed arguments to VPC peering function

The `is_peering_connection_pending` function as defined in
`modules/boto_vpc` is expecting a kwarg of `conn_name` but in the state
function `accept_vpc_peering_connection` on line 1483 it is called with an argument
of `name`. Updated the state function to use the proper kwarg.

Commit:
Merge pull request #36234 from rallytime/merge-carbon

[carbon] Merge forward from 2016.3 to carbon

rallytime commented:
Hi @blarghmatey - It looks like there are a bunch of commits in here that are not intended. Did you submit this against the correct branch?

twangboy committed:
Remove invalid JSON for Mac OS X (#36167)

* Remove invalid JSON for Mac OS X

* Check for fsevents line and remove

twangboy committed:
Add missing documentation for the mac_package module (#36192)

* Add missing documentation for the mac_package module

* Standardize ID in docstrings

terminalmage committed:
roots backend: Don't include '.' or '..' in empty_dirs

terminalmage committed:
Improve accuracy of fileserver runner tests

The tests that return files, symlinks, directories, and empty dirs were
all only testing that the type of the return data was the same as what
was expected. By not testing the content, we overlooked a corner case in
which backends passed into the fileserver as a Python list would not be
handled properly. This has since been fixed in PR #36244, but these
tests will help keep this sort of issue from regressing.

terminalmage committed:
Reduce fileserver_list_cache_time default to 20 sec

Commit:
fix archive test to only run on redhat

Commit:
Merge pull request #36248 from Ch3LL/fix_archive_test

fix archive test to only run on redhat

Commit:
Merge pull request #36247 from terminalmage/issue36150

Add a runner function to clear fileserver list caches

Commit:
Merge pull request #36176 from twangboy/mac_service_tests

Fix test_launchctl test in mac_service

blarghmatey commented:
@rallytime sorry about that. I just realized that I intended to issue this PR against the carbon branch and that's why the commit mismatch is there. Closing this PR and opening a new one against the correct branch.

rallytime commented:
@blarghmatey Thanks! I was thinking that might have been the case. :)
